controllers/__init__.py

```diff
@@ -47,14 +48,21 @@ def bootstrap_indexes(self) -> None:
                     f"Created DB index {keys}, collection: {collection}"
                 )
 
-    def get_circuits(self, archived: Optional[bool] = False) -> Dict:
+    def get_circuits(self, archived: Optional[bool] = False,
+                     metadata: dict = None) -> Dict:
         """Get all circuits from database."""
         aggregation = []
         match_filters = {"$match": {}}
+        aggregation.append(match_filters)
         if archived is not None:
             match_filters["$match"]["archived"] = archived
-            aggregation.append(match_filters)
-
+        if metadata:
+            for key in metadata:
+                if "metadata." in key:
```

Comment:
Shouldn't this only be matching if the key starts with "metadata."? Otherwise it'll give room to match more than it should like "foometadata.bar". 

Can you also verify if it's guaranteed if the keys will always be str or if we need to cast it?